packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1197,6 +1198,7 @@ class _TabBarViewState extends State<TabBarView> {
   TabController _controller;
   PageController _pageController;
   List<Widget> _children;
+  List<Widget> _originalChildrenWithKey;
```

Comment:
I presume all the changes regarding to _originalChildrenWithKey is due to the drag interruption?

Reply:
Yep, that's right.

Reply:
I remembered _originalChildrenWithKey is still needed while we don't support drag interruption. For instance, we're moving from 3rd tab to 1st tab with 10 seconds of duration, and if we **click** the 3rd tab, it still interrupts the animation. So we need to keep the original children in global.

Reply:
what would happen prior to this pr?

Reply:
Say we're moving from 3rd tab to 1st tab with 10 seconds. That means ```_childenWithKey``` (not ```_originalChildrenWithKey```) are mingled in ```_warpToCurrentIndex()```. While the animation is running, if I click the 3rd tab, we enter ```_warpToCurrentIndex()``` again. Prior to this pr, what would happen is we mingle up the already mingled up ```_childrenWithKey```, so when all the animations finish, ```_childrenWithKey``` settles in non-original order.

Reply:
Oops, I misunderstood your question. This was no problem prior to this pr because once animation is running inside ```_warpToCurrentIndex()```, it doesn't reenter that function until animation finishes (gets blocked in ```_handleTabControllerAnimationTick()```). However, now we're allowing it to reenter ```_warpToCurrentIndex()``` while animation is running (due to the changed condition in ```_handleTabControllerAnimationTick()```), so we need to keep the original children globally so they don't get messed up.

Reply:
If that is the case, why do we want to change the condition in `_handleTabControllerAnimationTick`? If we keep it the same, the clicking on tab will not interrupt the existing animation and it will not change the existing behavior of tab view.

Reply:
The thing is, if we click the tab, ```TabBar``` always gets interrupted. If we keep the condition in ```_handleTabControllerAnimationTick``` the same, only ```TabBarView``` will not get interrupted and it will look like below:
<img src="https://user-images.githubusercontent.com/4879766/87698657-bc9fc880-c7ce-11ea-9fc1-500014f806c7.gif" width=150/>

However by changing the condition, we make both of them move to the same destination:
<img src="https://user-images.githubusercontent.com/4879766/87699161-6e3ef980-c7cf-11ea-9b41-abe906a7057a.gif" width=150/>

Reply:
ah I see, so the tabbarview will not get interrupt while tabbar will. what if we make the tabbar uninterruptible? (by do a condition check in _handleTap)

Reply:
I can surely do that, but then user will have no way to interrupt the animation if it's taking long. Would that be okay?

Reply:
I believe that is the original intention but we only do it partially. I am open to make it interruptible, however, I can't think of a good way to make the animation to look ok(no weird jump or sudden content changes).

Reply:
I also think there's no way to make the interrupted animation look ok as of now. Then I'll make TabBar click also uninterruptible and let you know.